LINQ.Exercises/Orderby.cs

```diff
@@ -0,0 +1,217 @@
+            {
+                    new TestData.Person("Jean", "Gean", new DateTime(1950, 12, 1)),
+                    new TestData.Person("Jimmy", "Jilly", new DateTime(1974, 9, 16)),
+                    new TestData.Person("Jill", "Lill", new DateTime(2001, 5, 21)),
+                    new TestData.Person("Jack", "Tuck", new DateTime(1990, 3, 12))
+            }));
+        }
+
+        // hint: https://msdn.microsoft.com/en-us/library/bb549422(v=vs.110).aspx
+        // you need to create a class that implements the IComparer interface
+        // Strictly speaking, the MSDN documentation suggests that you should
+        // not inherit from IComparer, but your class should inherit from
+        // the Comparer<T> class. But for the purposes of this exercise
+        // inheriting from IComparer will do just fine.
+        [TestMethod]
+        public void OrderAlphabeticallyAssumingCaseInsensitivity_ReturnCaseInsensitiveEnumeration()
+        {
+            string[] words = { "aPPLE", "AbAcUs", "bRaNcH", "BlUeBeRrY", "ClOvEr", "cHeRry" };
+
+            // Вот здесь сортировка по-умолчанию сработала с IgnoreCase и тест пройден, НО
+            // решить это нужно было по-другому, создав свой собственный класс, реализовав в нем интерфейс
+            // IComparer (или IComparable?) или унаследоваться от класса Comparer
+            // И здесь у меня два вопроса:
+            //
+            // 1. Допустим я создал класс, унаследовав его от Comparer или реализовав один из интрефейсов,
+            // но как потом я должен был применить его? Ведь в сортируемом массиве хранятся данные строкового типа,
+            // а не выведенного мной. Мне нужно было в лямбда-выражении каким-то образом привести строковые элементы массива
+            // к моему пользовательскому типу и вернуть результат реализованного в пользовательском типе сравнения? Или как?
+            //
+            // 2. Так и не понял, что предпочтительней - наследование класса от Comparer, или реализация одного из интерфейсов
+            // (если релизация интерфейсов, то что предпочтительней Icomparer или Icomparable)? 
+            // Точнее мне не понятно почему предпочтительнее наследование нежели реализация. Все обсуждения этого вопроса на форумах, что я прочитал,
+            // цитируют все тот же MSDN, после которого я на эти форумы полез.
+            // Не мог бы ты мне пояснить почему именно наследование (в каких ситуациях это выгодно), и если наследование предпочтительнее, 
+            // то как быть в случаях, когда мне нужно наследовать свой класс от другого класса (естественно не Comparer),
+            // но в то же время необходимо реализовать функционал сравнения, ведь множественного наследования нет в C#.
+            // И если все же выбыирать реализацию интерфейса (опять же, когда это выгодно или необходимо?), то какой из них в каких случаях применять? 
+            var result = words.OrderBy(word => word);
```

Comment:
please implement your own `Comparer<string>`

Reply:
Will be done